microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/MicroProfilePropertiesListenerManager.java

```diff
@@ -171,6 +181,43 @@ public boolean visit(IResourceDelta delta) throws CoreException {
 			return false;
 		}
 
+		/**
+		 * Returns true if the given (folder) resource is in the Java output location
+		 * (ex : /target/classes) and false otherwise (ex: src/main/java).
+		 * 
+		 * @param resource the folder resource.
+		 * @return true if the given (folder) resource is in the Java output location
+		 *         (ex : /target/classes) and false otherwise (ex: src/main/java).
+		 * @throws JavaModelException
+		 */
+		private static boolean isInOutput(IResource resource) throws JavaModelException {
+
+			IJavaProject javaProject = JavaCore.create(resource.getProject());
+			IPath resourcePath = resource.getFullPath();
+
+			// Exclude resources in the main output location (e.g.,/ProjectName/bin)
+			IPath outputLocation = javaProject.getOutputLocation();
+			IPath outputFullPath = ResourcesPlugin.getWorkspace().getRoot().getFolder(outputLocation).getFullPath();

```

Comment:
This isn't the full path. 

You could compute the path relative to the workspace, in which case I don't think you need `outputFullPath`. you can do `outputLocation.isPrefixOf(resourcePath)`.

If you want the absolute path on the filesystem, why not do something like `javaProject.getProject().getLocation().append(outputLocation)` which should be the absolute path. I've seen it used in some other places.